src/main/scala/com/fortysevendeg/android/sqlite/util/ConnectionStringParser.scala

```diff
@@ -0,0 +1,39 @@
+package com.fortysevendeg.android.sqlite.util
+
+object ConnectionStringParser {
+
+  val urlRegex = "jdbc:sqlite:(:?[\\/\\.\\-_A-Za-z0-9]+)(\\?([A-Za-z0-9]+=[A-Za-z0-9]+)((\\&([A-Za-z0-9]+=[A-Za-z0-9]+))*)?)?".r
+
+  case class ConnectionValues(name: String, params: Map[String, String])
+
+  def parseConnectionString(connectionString: String): Option[ConnectionValues] = {
+    Option(connectionString) flatMap { c =>
+      val matcher = urlRegex.pattern.matcher(c)
+      matcher.matches() match {
+        case true =>
+          Some(ConnectionValues(
+            name = matcher.group(1),
+            params = readParams(matcher.group(3), matcher.group(4))))
+        case _ =>
+          None
+      }
+    }
+  }
+
+  private[this] def readParams(firstParam: String, lastParams: String): Map[String, String] =
+    (Option(firstParam), Option(lastParams)) match {
+      case (Some(f), Some(t)) if f.length > 0 => paramsToMap(f) ++ paramsToMap(cleanParams(t): _*)
+      case (Some(f), None) if f.length > 0 => paramsToMap(f)
+      case _ => Map.empty
+    }
+
+  private[this] def paramsToMap(params: String*): Map[String, String] =
+    Map(params map { p =>
+      val Array(name, value) = p.split("=")
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Field name does not match the regular expression '^[a-z][A-Za-z0-9]*$'](https://www.codacy.com/app/47deg/scala-sqlite-droid/file/809264727/issues/source?bid=2442814&fileBranchId=2738989#l32)